File: backend/app/models/project.py
from datetime import datetime
from uuid import UUID
from sqlalchemy import String, Boolean, ForeignKey, DateTime, Integer
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base, AuditMixin, SoftDeleteMixin

class Project(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "projects"
    
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    identifier: Mapped[str] = mapped_column(String(10), nullable=False)
    description: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    is_public: Mapped[bool] = mapped_column(Boolean, default=False)
    timezone: Mapped[str] = mapped_column(String(255), default="UTC")
    archived_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    
    workspace_id: Mapped[UUID] = mapped_column(ForeignKey("workspaces.id"), nullable=False)
    default_assignee_id: Mapped[UUID | None] = mapped_column(ForeignKey("users.id"), nullable=True)
    
    workspace: Mapped["Workspace"] = relationship(back_populates="projects")
    members: Mapped[list["ProjectMember"]] = relationship(back_populates="project")
    issues: Mapped[list["Issue"]] = relationship(back_populates="project")
    states: Mapped[list["State"]] = relationship(back_populates="project")
    cycles: Mapped[list["Cycle"]] = relationship(back_populates="project")
    labels: Mapped[list["Label"]] = relationship(back_populates="project")
    modules: Mapped[list["Module"]] = relationship(back_populates="project")
    attachments: Mapped[list["Attachment"]] = relationship(back_populates="project")
    notifications: Mapped[list["Notification"]] = relationship(back_populates="project")
    estimate_points: Mapped[list["EstimatePoint"]] = relationship(back_populates="project")
    issue_types: Mapped[list["IssueType"]] = relationship(back_populates="project")

class ProjectMember(Base, AuditMixin):
    __tablename__ = "project_members"
    
    project_id: Mapped[UUID] = mapped_column(ForeignKey("projects.id"), nullable=False)
    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
    role: Mapped[int] = mapped_column(Integer, default=15)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    
    project: Mapped["Project"] = relationship(back_populates="members")
    user: Mapped["User"] = relationship(back_populates="projects")

File: backend-go/internal/model/project.go
package model

import "time"

// Project represents a project within a workspace.
type Project struct {
	BaseModel

	Name              string     `gorm:"size:255;not null" json:"name"`
	Identifier        string     `gorm:"size:10;not null" json:"identifier"`
	Description       *string    `gorm:"size:1000" json:"description"`
	IsPublic          bool       `gorm:"default:false" json:"is_public"`
	Timezone          string     `gorm:"size:255;default:UTC" json:"timezone"`
	ArchivedAt        *time.Time `json:"archived_at"`
	WorkspaceID       uint64     `gorm:"not null;index" json:"workspace_id"`
	DefaultAssigneeID *uint64    `json:"default_assignee_id"`
	Color             string     `gorm:"size:20;default:'#6366F1'" json:"color"`
	TemplateID        *uint64    `json:"template_id"`

	// Relationships
	Workspace       Workspace       `gorm:"foreignKey:WorkspaceID" json:"-"`
	DefaultAssignee *User           `gorm:"foreignKey:DefaultAssigneeID" json:"-"`
	Members         []ProjectMember `gorm:"foreignKey:ProjectID" json:"-"`
	Issues          []Issue         `gorm:"foreignKey:ProjectID" json:"-"`
	States          []State         `gorm:"foreignKey:ProjectID" json:"-"`
	Labels          []Label         `gorm:"foreignKey:ProjectID" json:"-"`
	Cycles          []Cycle         `gorm:"foreignKey:ProjectID" json:"-"`
}

func (Project) TableName() string {
	return "projects"
}

// ProjectMember represents a user's membership in a project.
type ProjectMember struct {
	BaseModel

	ProjectID uint64 `gorm:"not null;uniqueIndex:idx_proj_member_user" json:"project_id"`
	UserID    uint64 `gorm:"not null;uniqueIndex:idx_proj_member_user" json:"user_id"`
	Role      int    `gorm:"default:15" json:"role"`
	IsActive  bool   `gorm:"default:true" json:"is_active"`

	// Relationships
	Project Project `gorm:"foreignKey:ProjectID" json:"-"`
	User    User    `gorm:"foreignKey:UserID" json:"-"`
}

func (ProjectMember) TableName() string {
	return "project_members"
}
